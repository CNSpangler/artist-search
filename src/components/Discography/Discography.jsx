import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Discography = ({ artistName, releases }) => {
  const releaseElements = releases.map(release => (
    <li key={release.releaseId}>
      <Link to={`/releases/${artistName}/${release.releaseId}`}>
        <img src={`http://coverartarchive.org/release/${release.releaseId}/front`} />
        <span>{release.releaseTitle}: {release.releaseDate}</span>
      </Link>
    </li>
  ));

  return (
    <ul>
      {releaseElements}
    </ul>
  );
};

Discography.propTypes = {
  artistName: PropTypes.string.isRequired,
  releases: PropTypes.arrayOf(PropTypes.shape({
    releaseId: PropTypes.string.isRequired,
    releaseTitle: PropTypes.string.isRequired,
    releaseDate: PropTypes.string.isRequired
  }))
};

export default Discography;
