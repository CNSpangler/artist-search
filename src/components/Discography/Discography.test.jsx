import React from 'react';
import { shallow } from 'enzyme';
import Discography from './Discography.jsx';

describe('Discography component', () => {
  it('renders Discography', () => {
    const wrapper = shallow(<Discography releases={[
      {
        releaseId: 1234,
        releaseTitle: 'Jock Jamz',
        releaseDate: 1985
      },
      {
        releaseId: 1234,
        releaseTitle: 'Jock Jamzz',
        releaseDate: 1986
      }
    ]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
