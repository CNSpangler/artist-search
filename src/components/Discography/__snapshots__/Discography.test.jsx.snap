// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Discography component renders Discography 1`] = `
<ul>
  <li
    key="1234"
  >
    <Link
      to="/releases/1234"
    >
      <img
        src="http://coverartarchive.org/release/1234/front"
      />
      <span>
        Jock Jamz
        : 
        1985
      </span>
    </Link>
  </li>
  <li
    key="1234"
  >
    <Link
      to="/releases/1234"
    >
      <img
        src="http://coverartarchive.org/release/1234/front"
      />
      <span>
        Jock Jamzz
        : 
        1986
      </span>
    </Link>
  </li>
</ul>
`;
